session/session_store.go

```diff
@@ -85,18 +89,27 @@ func (r *RedisStore) RegisterEndpoint(s Session) error {
 	defer redisConn.Close()
 
 	redisConn.Send("MULTI")
-	redisConn.Send("HSET", s.Key(), "endpoint_addr", s.Endpoint())
 	redisConn.Send("HSET", s.Key(), "cluster", s.Cluster())
-	redisConn.Send("SADD", "backend:"+s.BackendID()+":endpoints", s.Endpoint())
-	endpoint := map[string]string{
-		"session_id":   s.ID(),
-		"backend_id":   s.BackendID(),
-		"socket":       s.Endpoint(),
-		"cluster":      s.Cluster(),
-		"connected_at": t.String(),
-		"last_seen_at": t.String(),
+	for _, endpointAddr := range s.Endpoints() {
+		redisConn.Send("SADD", "backend:"+s.BackendID()+":endpoints", redisEndpointString(endpointAddr))
+		endpoint := map[string]string{
+			"session_id":   s.ID(),
+			"backend_id":   s.BackendID(),
+			"cluster":      s.Cluster(),
+			"connected_at": t.String(),
+			"last_seen_at": t.String(),
+		}
+
+		if extended, ok := endpointAddr.(wnet.ExtendedAddr); ok {
+			switch t := extended.Data().(type) {
+			case wnet.SingleTLSAddrExtendedData:
```

Comment:
I'm  wanting opinions here. We could either type check against any additional extended data types. Or... We could structure the ExtendedData structs in ./net/listener* such that we could just `AddFlat(extended.Data())` Here. 

Tradeoffs I see are, if we keep the way it is implemented now, we have fine-grained control of the data going into redis at the stowage point - but weakening the `wnet.ExtendedAddr` abstraction.

If we want to keep the `wnet.ExtendedAddr` abstraction completely opaque, then we could ensure that the interface returned by `extended.Data()` could be safely be passed direct to `AddFlat()` - and then this could reduce the `SessionStore` abstraction since `wnet.ExtendedAddr` data types would need to cater to redis-safe datatypes

Thoughts?

Reply:
Hmm... it definitely feels like `wnet` package is unnecessarily "polluting" this file. On the other hand, we only have (and likely going to have for the foreseeable future) `SingleTLSAddrExtendedData` that only has one field: `ca_cert` (and having the cert chain here is also questionable, since we're only going to use one cert - at most - per wormhole server for ingress connections).

I can see that we might want to bake the authentication in - like client cert authentication - so we might have a few more fields for that particular endpoint type.

I'd say, I'm not opposing having this additional code here and tackle this later, if it makes our lives difficult at any point.

Reply:
My vote would be leave it here now. And perhaps later do a fix. I don't think a decision needs to be made now.